qt/scientific_interfaces/Inelastic/Analysis/ConvFitDataView.cpp

```diff
@@ -33,15 +33,19 @@ ConvFitDataView::ConvFitDataView(const QStringList &headers, QWidget *parent) :
   header->setSectionResizeMode(1, QHeaderView::Stretch);
 }
 
-IAddWorkspaceDialog *ConvFitDataView::getAddWorkspaceDialog() {
-  m_addWorkspaceDialog = new ConvFitAddWorkspaceDialog(parentWidget());
-  if (auto dialog = dynamic_cast<ConvFitAddWorkspaceDialog *>(m_addWorkspaceDialog)) {
-    dialog->setResolutionWSSuffices(m_wsResolutionSuffixes);
-    dialog->setResolutionFBSuffices(m_fbResolutionSuffixes);
-  }
-  connect(m_addWorkspaceDialog, SIGNAL(addData()), this, SLOT(notifyAddData()));
-
-  return m_addWorkspaceDialog;
+void ConvFitDataView::showAddWorkspaceDialog() {
+  auto dialog = new ConvFitAddWorkspaceDialog(parentWidget());
+  connect(dialog, SIGNAL(addData()), this, SLOT(notifyAddData()));
+
+  dialog->setAttribute(Qt::WA_DeleteOnClose);
+  dialog->setWSSuffices(m_wsSampleSuffixes);
+  dialog->setFBSuffices(m_fbSampleSuffixes);
+  dialog->setResolutionWSSuffices(m_wsResolutionSuffixes);
+  dialog->setResolutionFBSuffices(m_fbResolutionSuffixes);
+  dialog->updateSelectedSpectra();
+  dialog->show();
+
+  m_addWorkspaceDialog = dialog;
```

Comment:
This is just a question: I know lifetime of dialog is handled by qt classes, but seems you can create as many workspacedialogs as you want and the member `m_addWorkspaceDialog` would be pointing to the last one created. Can't that be a problem if multiple dialog instances are open at the same time and you try to add data from, say, the first opened workspaceDialog?

Reply:
Good spot, you are right, this will cause an issue. I haven't fixed this yet in the latest commits - I will fix it when I get back from A/L, so lets keep this PR open until then

Reply:
This is also a problem on `main` and `release-next`, so I have opened an issue #36756 that can be tracked for the current release cycle. This PR fixes the problem, so we should maybe consider for this to go into release-next

Reply:
I agree.